authorization/invitation/service/invitation_service.go

```diff
@@ -344,72 +360,80 @@ func (s *invitationServiceImpl) Rescind(ctx context.Context, rescindingUserID, i
 }
 
 // Accept processes an invitation acceptance click, and returns the resource ID of the resource or identity resource which the invitation is for
-func (s *invitationServiceImpl) Accept(ctx context.Context, token uuid.UUID) (string, error) {
-	var resourceID string
+func (s *invitationServiceImpl) Accept(ctx context.Context, token uuid.UUID) (string, string, error) {
+	var resourceID, redirectURL string
 
 	// Locate the invitation
 	inv, err := s.Repositories().InvitationRepository().FindByAcceptCode(ctx, token)
 
 	if err != nil {
-		return resourceID, err
+		return resourceID, redirectURL, err
 	}
 
+	// get redirect urls
+	redirectURL = inv.SuccessRedirectURL
+	redirectOnFailure := inv.FailureRedirectURL
+
 	// get identity for invitation
 	currentIdentityID := inv.IdentityID
-	identity, e := s.Repositories().Identities().LoadWithUser(ctx, inv.IdentityID)
-	if e != nil {
-		return resourceID, err
+	identity, err := s.Repositories().Identities().LoadWithUser(ctx, currentIdentityID)
+	if err != nil {
+		return resourceID, redirectOnFailure, errs.Wrap(err, fmt.Sprintf("failed to load identity for invitee %d", currentIdentityID))
 	}
+
 	if identity.User.Deprovisioned {
-		return resourceID, autherrors.NewUnauthorizedError("user deprovisioined")
+		return resourceID, redirectOnFailure, autherrors.NewUnauthorizedError("user deprovisioned")
 	}
 
+	invitationID := inv.InvitationID
 	// If this invitation is for an identity
 	if inv.InviteTo != nil {
-		inviteToIdentity, err := s.Repositories().Identities().Load(ctx, *inv.InviteTo)
+		inviteTo := *inv.InviteTo
+		inviteToIdentity, err := s.Repositories().Identities().Load(ctx, inviteTo)
 		if err != nil {
-			return resourceID, err
+			return resourceID, redirectOnFailure, errs.Wrap(err, fmt.Sprintf("failed to load identity for inviteTo %d", inviteTo))
 		}
 
 		// If the invitation is for a membership, add a membership record
 		if inv.Member {
 			s.Repositories().Identities().AddMember(ctx, inviteToIdentity.ID, currentIdentityID)
 		}
 
-		roles, err := s.Repositories().InvitationRepository().ListRoles(ctx, inv.InvitationID)
+		roles, err := s.Repositories().InvitationRepository().ListRoles(ctx, invitationID)
```

Comment:
can't we load the invitation with the roles in a single repository call (using the `db.Preload()`) function ?

Reply:
no

Reply:
why not ?

Reply:
let's discuss it in 1 x 1